src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private Integer supplySum;
+    private Integer buySum;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File fromFile = new File(fromFileName);
+        File toFile = new File(toFileName);
+        getSum(fromFile);
+        writeData(toFile);
+    }
+
+    private void getSum(File file) {
+        buySum = 0;
+        supplySum = 0;
+        try (FileReader fileReader = new FileReader(file);
+                BufferedReader reader = new BufferedReader(fileReader)) {
+            while (reader.ready()) {
+                String[] array = reader.readLine().split(",");
+                if (array[0].equals("buy")) {
+                    buySum += Integer.parseInt(array[1]);
+                }
+                if (array[0].equals("supply")) {
+                    supplySum += Integer.parseInt(array[1]);
+                }
+            }
```

Comment:
Extract `read` and `calculate` operations to the separate methods. Call it from `getStatistic`. Use returns instead of global variables.